### ISSUE TITLE
Please help making this run again

### ISSUE THREAD

**Issue by Bararun** [mrmin123/kancolle-auto#105]
After a year long hyatus I go back to Kancolle and have taken a look at Kancolle-Auto again. Read all the guidelines and installed all the neccesary packages. I try running the script with quests and expeditions enabled and after 2 random cursor movements around the game screen the cursor stops for 4-5 secs and this error pops up:

[error] script [ kancolle_auto ] stopped with error in line 377
[error] AttributeError ( 'module' object has no attribute 'Expedition' )
[error] --- Traceback --- error source first line: module ( function ) statement 346: main ( init ) expedition_item = expedition_module.Expedition(kc_window, settings)
346: main ( init ) expedition_item = expedition_module.Expedition(kc_window, settings)

Been already 4 hours already trying to descipher what is exactly happening but no luck yet. I'll keep trying tomorrow (4:00 AM here) but if you can tell me what could be wrong I'd apreciate it. Just in case I tell you I use Win10, I hope this isn't the root of the problem.

Thank you for your work doing this script and for your time helping me with this issue

**Comment by mrmin123** [comment]
Hi @Bararun, sorry to hear you're having issues. I unfortunately don't have a Win10 machine so I can't test Sikuli compatibility (some discussion regarding that located [here](https://answers.launchpad.net/sikuli/+question/269810)). That said, you mention you had both Quests and Expeditions modules enabled, but the script isn't crashing at Quests module initialization, which occurs before the Expedition module initialization, so I'm wondering if there's something wrong with the file structure. Does the `expedition.sikuli` directory exists under `kancolle-auto.sikuli`, and does `expedition.py` exists in that folder?

**Comment by mrmin123** [comment]
Also, try deleting any `*.pyc` files that were auto-generated before re-running. These compiled files can overwrite any changes you've made to the `*.py` source files on execution.

**Comment by Bararun** [comment]
Yes indeed, everything seems to be in order related with what you have described. Where should I look for those .pyc files?

**Comment by mrmin123** [comment]
They should be located next to the `*.py` files... That said, if you never changed the source files, removing the `*.pyc` files shouldn't have an effect.

**Comment by n0k0m3** [comment]
Tested on Win 10, working ok...

Can you provide a screenshot + flash player renderer setting?

**Comment by Bararun** [comment]
There are no .pyc files generated inside kancolle_auto.sikuli so there was nothing for me to delete.
Flash setting + screen http://i.imgur.com/rbdz7uZ.png

**Comment by n0k0m3** [comment]
Can you copy a full log from the start with "-d 2" run option?

**Comment by Bararun** [comment]
I guess you are telling me to run this on cmd:  java -jar Path\sikuli-script.jar -r Path\kancolle_auto.sikuli -d 2

**Comment by Bararun** [comment]
BTW I usually run the script from the IDE, but I've already tried using the cmd too with identical results

**Comment by n0k0m3** [comment]
Yep

**Comment by n0k0m3** [comment]
Just post the full log here and I'll check what's wrong

**Comment by mrmin123** [comment]
Also, do you remember which build/version you're using?

**Comment by Bararun** [comment]
I've tried both r2 and r2.1 with identical results. I think this log is from r2.1 since the afected line is the 375

Full log 

C:\BLABLA>java -jar C:Path\sikuli-script.jar -r C:Path\kancolle-auto\kancolle_auto.sikuli -d 2
ene 23, 2016 5:40:46 AM java.util.prefs.WindowsPreferences <init>
WARNING: Could not open/create prefs root node Software\JavaSoft\Prefs at root 0x80000002. Windows RegCreateKeyEx(...) returned error code 5.
[debug] Running on Java 8 (1.8.0_71-b15)
[debug] ResourceLoaderBasic: check: we are running on arch: amd64
[debug] ResourceLoaderBasic: check: using Java at: C:/Program Files/Java/jre1.8.0_71/
[debug] Screen: initScreens: basic initialization (1 Screen(s) found)
[2016-01-23 05:40:53] Reading config file
[2016-01-23 05:40:53] Expeditions (fleet 2: 2, fleet 3: 5, fleet 4: 38) enabled!
[2016-01-23 05:40:53] Config loaded!
[2016-01-23 05:40:53] Reading config file
[2016-01-23 05:40:53] Starting kancolle_auto
[2016-01-23 05:40:53] Finding window!
[2016-01-23 05:40:53] Focus on KanColle!
[log] App.focus KanColleViewer!(0) #0
[log] App.focus KanColleViewer!(0) #0
[2016-01-23 05:41:08] Defining module items!
[2016-01-23 05:41:08] Quests reset. Checking for the following quests: ['d11', 'd2', 'd3', 'd4', 'd9']
[error] script stopped with error in line 375
[error] AttributeError ( 'module' object has no attribute 'expedition' )
[error] --- Traceback --- error source first
line: module ( function ) statement
344: main (  init )     expedition_item = expedition_module.expedition(kc_window, settings)
344: main (  init )     expedition_item = expedition_module.expedition(kc_window, settings)
[error] --- Traceback --- end --------------

C:\BLABLA>

**Comment by Bararun** [comment]
nope, this log is r2

Edit: And now Akashi is inviting me to Mamiya's to have breakfast...

**Comment by n0k0m3** [comment]
It seemed I cannot reproduce the error tho...
BTW, is that true that ur putting the script in OS partition?
If then, move it to other partition and try again, probably the script cannot import the module because it doesn't have enough privileges to access files on OS partition.

**Comment by mrmin123** [comment]
@Bararun can you insert this code snippet:

```
print 'TEST'
for name, obj in inspect.getmembers(expedition_module):
    if inspect.isclass(obj):
        print obj
print 'TEST'
```

before the first `init()` call (where it says `# initialize kancolle_auto`) and post what shows up in console between the 'TEST' lines here?

**Comment by Bararun** [comment]
@minh6a Just to be sure, could it be that since I'm using the same Sikuli 1.0.1 from a year ago I could be having this problem? I figured I didn't need to look for the build again since it was the same version (?)

Unfortunately I only have the OS partition in this laptop. Nevertheless, I ran Kancolle-auto a year ago with the same paths without problems. Furthermore, if the problem were the privileges, wouldn't the quests module have problems to load too?

@mrmin123 ran it through the IDE and this shows up:

TEST
expedition.Ensei
TEST

**Comment by Bararun** [comment]
BTW, that wasn't the first init() in the script. it is at the line 332

**Comment by mrmin123** [comment]
Something's definitely up with how the expedition module is being loaded... my local version:

```
TEST                                                      
<type 'org.sikuli.script.App'>                            
<type 'org.sikuli.script.Button'>                         
<type 'org.sikuli.basics.Debug'>                          
<type 'org.sikuli.script.compare.DistanceComparator'>     
expedition.Ensei                                          
<class 'sikuli.Env.Env'>                                  
expedition.Expedition                                     
<type 'org.sikuli.basics.ExtensionManager'>               
<type 'org.sikuli.script.FindFailed'>                     
<type 'org.sikuli.script.Finder'>                         
<type 'org.sikuli.script.compare.HorizontalComparator'>   
<type 'org.sikuli.basics.HotkeyListener'>                 
<type 'org.sikuli.basics.HotkeyManager'>                  
<type 'org.sikuli.basics.ImageLocator'>                   
<type 'org.sikuli.script.Env'>                            
<type 'org.sikuli.script.Region'>                         
<type 'org.sikuli.script.Screen'>                         
<type 'org.sikuli.script.Key'>                            
<type 'org.sikuli.script.KeyModifier'>                    
<type 'org.sikuli.script.Location'>                       
<type 'org.sikuli.script.Match'>                          
<type 'org.sikuli.basics.OS'>                             
<type 'org.sikuli.script.Pattern'>                        
<class 'sikuli.Region.Region'>                            
<class 'sikuli.Screen.Screen'>                            
<type 'org.sikuli.script.ScreenUnion'>                    
<type 'org.sikuli.basics.Settings'>                       
<type 'org.sikuli.script.SikuliEventAdapter'>             
<type 'org.sikuli.basics.SikuliScript'>                   
<type 'org.sikuli.script.SikuliX'>                        
<type 'org.sikuli.script.compare.VerticalComparator'>     
util.color                                                
<type 'time'>                                             
TEST                                                      
```

**Comment by Bararun** [comment]
Ok seems that this may be relevant too. I've been having this error but I thought until now that it was due to me doing a bad imput of the paths but after running the test once, getting the usual expedition module loading problems, etc. Now I'm having this.

TEST
expedition.Ensei
TEST
[94m[2016-01-23 06:26:39] Reading config file[0m
[92m[2016-01-23 06:26:39] Expeditions (fleet 2: 2, fleet 3: 5, fleet 4: 38) enabled![0m
[92m[2016-01-23 06:26:39] Config loaded![0m
[94m[2016-01-23 06:26:39] Reading config file[0m
[92m[2016-01-23 06:26:39] Starting kancolle_auto[0m
[94m[2016-01-23 06:26:39] Finding window![0m
[94m[2016-01-23 06:26:39] Focus on KanColle![0m
[log] App.focus KanColleViewer!(0) #0
[error] script [ kancolle_auto ] stopped with error in line 382
[error] AttributeError ( 'NoneType' object has no attribute 'x' )

**Comment by Bararun** [comment]
tried copying the build from the zip file again, Went back to " 'module' object has no attribute 'Expedition' ".
Inserted the snipet again and again " 'module' object has no attribute 'Expedition' "
Looks like everything is back to how it "should be", at least.

PD: Same test message BTW.

**Comment by mrmin123** [comment]
I'm at a loss. I've never heard of only a specific subset of classes from a module being loaded. My only suggestions right now is to 1. nuke the kancolle-auto folder (back up your config.ini, delete everything, extract the build from zip again, restore config.ini, try again), and then 2. nuke and re-install Sikuli (although I really doubt this will changed anything as well)

**Comment by mrmin123** [comment]
Your Python is 2.7.x, right?

**Comment by Bararun** [comment]
Alleluyah! nuked all the kancolle-auto folders I had, downloaded r2.1 again, set up everything through the IDE and now it suddenly works... for now, since that new "paranoia" feature is hilariously annoying. At least it seems it loads the modules.

And yeah, my Python is 2.7.11

Sorry for all the inconveniences. I'd really like to give you some hints about why this happened but I cant fathom why, at least from my limited point of view.

Thank you very much!

**Comment by mrmin123** [comment]
No problem! Glad you got it to work again! If you don't like the Paranoia feature you can disable it by setting it to 0 in the config. Enjoy!

**Comment by mrmin123** [comment]
Also, I just realized, if you just deployed my build of kancolle-auto over another kancolle-auto build from a year ago, this might've caused the issue. My iteration of the script is vastly different from the predecessor and are not compatible in the slightest.

**Comment by Bararun** [comment]
I did indeed copy the new build in the folder of the older build I had from one year ago the first time I tried to make the script run. Then I also thought what you just said and used different folders, new folders directly created from decompresing the build from .zip

But, in any case, I allways just replaced all files with the new ones wether on the old build folder or the new folders. Wouldn't that had bypassed that problem?

**Comment by mrmin123** [comment]
When replacing directly, some files might not have gotten overwritten, as I've moved many of the files and modules around. Also, if you had old `*.pyc` files remaining in the subfolders, they would take precedence over the new modularized `*.py` files. This is actually the reason I brought that up... I remembered that the original `expedition.py` only had the `Ensei` class build into it (what you were seeing in your TEST output), and my new version expanded on it with the `Expedition` class and importing of all the Sikuli classes.
